quinn/src/endpoint.rs

```diff
@@ -452,22 +447,23 @@ impl State {
                                         .send(ConnectionEvent::Proto(event));
                                 }
                                 Some(DatagramEvent::Response(transmit)) => {
-                                    // Limiting the memory usage for items queued in the outgoing queue from endpoint
-                                    // generated packets. Otherwise, we may see a build-up of the queue under test with
-                                    // flood of initial packets against the endpoint. The sender with the sender-limiter
-                                    // may not keep up the pace of these packets queued into the queue.
-                                    if self.transmit_queue_contents_len
-                                        < MAX_TRANSMIT_QUEUE_CONTENTS_LEN
-                                    {
-                                        let contents_len = transmit.size;
-                                        self.outgoing.push_back(udp_transmit(
+                                    // Send if there's kernel buffer space; otherwise, drop it
```

Comment:
Ugh, I forgot everything about what this is. What are endpoint-generated datagrams used for? Maybe add some comments for context?

Reply:
Added substantially more detail.